meta-balena-common/recipes-containers/update-data-store/files/update-data-store

```diff
@@ -0,0 +1,285 @@
+#!/bin/bash
+
+set -e
+
+# shellcheck disable=SC1091
+. /usr/libexec/os-helpers-logging
+
+ENGINE_OVERLAY_ROOT="/var/lib/docker/overlay2/"
+
+# Detect containers engine
+if which balena > /dev/null 2>&1; then
+    ENGINE=balena
+else
+    error "No container engine detected."
+fi
+
+readonly script_name=$(basename "${0}")
+usage() {
+	cat <<EOF
+Usage: ${script_name} [OPTIONS]
+	-t Target state json
+	-r Reboot after update
+EOF
+	exit 0
+}
+
+error_handler() {
+	# shellcheck disable=SC2181
+	[ $? -eq 0 ] && exit 0
+	exit 1
+}
+
+trap error_handler EXIT
+
+RESIN_DATA_DIR="/mnt/data"
+APPS_JSON_PATH="${RESIN_DATA_DIR}/apps.json"
+
+verify_target_images() {
+	local _target_images="${1}"
+	local _image
+	local _imageid
+	for _image in ${_target_images}; do
+		_imageid=$(${ENGINE} inspect -f '{{.Id}}' "${_image}" 2> /dev/null) || _imageid=""
+		if [ -n "${_imageid}" ]; then
+			continue
+		fi
+
+		if ! DOCKER_CLI_EXPERIMENTAL=enabled ${ENGINE} manifest inspect "${_image}"  > /dev/null 2>&1; then
+			error "Not found in repository - bailing out with no action taken: ${_image}"
+			return 1
+		fi
+	done
+}
+
+umount_merged() {
+	local _cid="${1}"
+	local _mergeddir
+	local _overlay
+	_mergeddir=$(${ENGINE} inspect -f '{{.GraphDriver.Data.MergedDir}}' "${_cid}" 2> /dev/null) || _mergeddir=""
+	if [ -n "${_mergeddir}" ]; then
+		_mergeddir=${_mergeddir#${ENGINE_OVERLAY_ROOT}}
+		_overlay=$(findmnt --noheadings --canonicalize  -t overlay -l -o TARGET | grep "${_mergeddir}" || true)
+		if [ -n "${_overlay}" ]; then
+			umount -l "${_overlay}" || true
+		fi
+	fi
+}
+
+remove_data_overlays() {
+	local _target_images="${1}"
+	local _image
+	local _oldimage
+	local _imagetag
+	local _cids
+	local _cid
+	info "Removing previous data store images"
+	for _image in $(${ENGINE} images --all --quiet --filter label="${BALENA_HOSTOS_BLOCK_STORE}=data" --filter label="${BALENA_HOSTOS_BLOCK_CLASS}=overlay"); do
```

Comment:
I think variables `BALENA_HOSTOS_BLOCK_STORE`, `BALENA_HOSTOS_BLOCK_CLASS` and `BALENA_HOSTOS_BLOCK_REQUIRES_REBOOT` are not being defined anywhere.

Reply:
They are now defined in https://github.com/balena-os/meta-balena/blob/741778f692347c812e1a682cad83dba73069ed23/meta-balena-common/recipes-support/os-helpers/os-helpers/os-helpers-api#L4 as they are used in both `update-data-store` and the `docker-disk` recipe,